gst/nnstreamer/nnstreamer_plugin_api_impl.c

```diff
@@ -923,6 +923,83 @@ _peer_is_flexible_tensor_caps (GstPad * pad)
   return flexible;
 }
 
+/**
+ * @brief Update caps dimension for negotiation
+ * @param caps caps to compare and update
+ * @param peer_caps caps to compare
+ */
+void
+update_caps_dimension (GstCaps * caps, GstCaps * peer_caps)
+{
+  GstStructure *structure;
+  GstStructure *structure_peer;
+  tensor_dim dim, dim_peer;
+  const gchar *dim_str;
+  const gchar *dim_str_peer;
+  guint i;
+
+  if (gst_caps_get_size (caps) == 0 || gst_caps_get_size (peer_caps) == 0)
+    return;
+
+  structure = gst_caps_get_structure (caps, 0);
+  dim_str = gst_structure_get_string (structure, "dimension");
+  gst_tensor_parse_dimension (dim_str, dim);
```

Comment:
dim_str can be null in some cases if there is no dimension in structure.
how about check retrun value ?
if (!gst_tensor_parse_dimension (dim_str, dim)) {
  /* error log */
  return;
}

Reply:
Updated to parse dimension when structure has dimension properties only.
Thank you!